src/dbm.c

```diff
@@ -241,9 +241,8 @@ bool dbm_intersection(raw_t* dst, const raw_t* src, cindex_t dim)
 
     if (dim > 1) {
         cindex_t i, j, ci = 0, cj = 0, count = 0;
-        uint32_t touched[bits2intsize(dim)]; /* to reduce final close */
+        uint32_t* touched = (uint32_t*)calloc(bits2intsize(dim), sizeof(uint32_t));
```

Comment:
this will have very serious performance issues: malloc/calloc is very expensive -- I really do not like this.
In C++ I would recommend `std::pmr`, in C we could do the same: allocate some fixed amount on stack and allocate dynamically if exceeding some maximum. Then we need to figure out a sensible maximum.

Reply:
consider:
```c
#define MAX_STACK 1000
void compute(size_t size){
	int* mem = (size > MAX_STACK) ? (int*)malloc(size*sizeof(int)) : (int*)alloca(size*sizeof(int));
	mem[0] = 1;
	mem[1] = 1;
	for (int i = 2; i<size; ++i)
		mem[i] = mem[i-1] + mem[i-2];
	for (int i = 0; i<size; ++i)
		printf("%d ", mem[i]);
	if (size > MAX_STACK)
		free(mem);
}
```

Reply:
refined example:
```c
#include <stdio.h>  // printf
#include <stdlib.h> // malloc
#include <stdbool.h> // bool
#include <alloca.h>

#define MAX_STACK 1000
#define fast_alloc(ptr, size) mem = (size > MAX_STACK) ? malloc(size*sizeof(*ptr)) : alloca(size*sizeof(*ptr));
#define fast_free(ptr, size) if (size > MAX_STACK) free(ptr);

void compute(size_t size){
	int* mem;
	fast_alloc(mem, size);
	mem[0] = 1;
	mem[1] = 1;
	for (int i = 2; i<size; ++i)
		mem[i] = mem[i-1] + mem[i-2];
	for (int i = 0; i<size; ++i)
		printf("%d ", mem[i]);
	fast_free(mem, size);
}

int main(){
	compute(40);
}
```

Reply:
I do understand what you're proposing.
But I'd like to have a benchmark that shows that alloca gives us a significant performance gain above malloc before we begin implementing the funky macro that calls alloca, which is non-standard.

I've set up a shell script that runs all the working unit tests with some pretty high values, and this time, the implementation that uses malloc actually did slightly better.
Maybe these VLAs don't actually have that much of an importance in regards to performance.

So unless we can find a benchmark that shows that the malloc branch is slower, I think it should be merged. We shouldn't optimize performance based on gut feeling.